r/vignettes/developers/setup.Rmd

```diff
@@ -40,13 +40,15 @@ set -x
 
 The Arrow R package is unique compared to other R packages that you may have
 contributed to because it builds on top of the large and feature-rich Arrow C++
-implementation.
+implementation. Because the R package integrates tightly with Arrow C++,
+it typically requires a dedicated copy of the library (i.e., it is usually
+not possible to link to a system version of libarrow during development).
 
-## Option 1: Using the bundled build with nightly libarrow binaries
+## Option 1: Using nightly libarrow binaries
 
-On Linux and MacOS, you can use the same workflow you might use for another
-package that contains compiled code (e.g., `R CMD INSTALL . --preclean` from
```

Comment:
I am not sure but I think not having --preclean will prevent recompilation of the R files even when a new arrow version was fetched?